### ISSUE TITLE
v3.x/Psalm 6: init's runTaintAnalysis=true silently disables type analysis; contradicts add workflow + baseline

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1139]
## Summary

On the **v3.x line (Psalm 6)**, `psalm-laravel init` and the `add` CI workflow contradict each other, and the result is that **type analysis silently never runs**.

Found while onboarding the plugin onto [cachethq/core](https://github.com/cachethq/core) (Laravel 11, `psalm/plugin-laravel v3.14.1`, `vimeo/psalm 6.16.1`).

## 1. `init` writes `runTaintAnalysis="true"`, which disables type analysis on Psalm 6

`vendor/bin/psalm-laravel init` generates a `psalm.xml` with `runTaintAnalysis="true"`. On Psalm 6, `--taint-analysis` / `runTaintAnalysis` switches Psalm into a **taint-only mode that skips type analysis**. So a plain `vendor/bin/psalm` run reports **zero type issues** — the type gate is silently dead.

### Reproduction

With the generated config (`runTaintAnalysis="true"`), add a file with an obvious type error:

```php
<?php
namespace App;
final class Probe { public function run(): int { return "definitely a string"; } }
```

```
$ vendor/bin/psalm
No errors found!
```

Remove `runTaintAnalysis="true"` from the config and re-run:

```
ERROR: InvalidReturnType ... The declared return type 'int' ... is incorrect, got ''definitely a string''
ERROR: InvalidReturnStatement ...
```

A developer running plain `vendor/bin/psalm` locally gets a green run that did **no type checking**.

## 2. `init` config contradicts the `add` CI workflow

The two-job workflow (`resources/ci/github-actions/psalm.yml`) runs the **type job** as plain `./vendor/bin/psalm`, with a header comment stating *"Psalm 6.x does not run taint by default"* — i.e. it assumes the config has **no** `runTaintAnalysis`. But `init` emits exactly that attribute. Used together, **both** jobs run in taint-only mode and type analysis never executes in CI.

## 3. A type baseline then breaks the taint job (`UnusedBaselineEntry`)

Once a type baseline is wired in (`errorBaseline="psalm-baseline.xml"`, as `--set-baseline` writes), the **taint job** (`--taint-analysis`) loads it, never recomputes the type issues, and reports every entry as `UnusedBaselineEntry`. On cachet/core that was **168 errors**, failing the taint job. The template's dual-baseline note (`psalm-baseline.xml` + `psalm-taint-baseline.xml`) is internally inconsistent on Psalm 6: there is only one `errorBaseline` attribute, and `--set-baseline` for the taint mode overwrites it.

## Suggested fixes

- On the v3.x line, **`init` should not emit `runTaintAnalysis="true"`** (let the `add` workflow drive taint via the `--taint-analysis` flag in its dedicated job). This keeps `init` and `add` consistent and keeps a plain local `vendor/bin/psalm` doing type analysis.
- The generated **taint job should run with `--ignore-baseline`** so a type baseline does not surface as `UnusedBaselineEntry` (or document a separate taint baseline that is actually wired per-job).

Both fixes were applied by hand in the cachet/core onboarding PR and verified: type gate green (baseline applied), taint gate green (0 SARIF).

## Notes

This is **v3.x / Psalm 6 specific**. On Psalm 7 / v4.x, taint and type run in a single pass, so the `runTaintAnalysis` attribute is harmless there.

**Comment by alies-dev** [comment]
Update: reproduced on the **latest** plugin release `v4.12.3` as well (it requires `vimeo/psalm ^6.16.1`, and 6.16.1 is the newest 6.x, so latest-installable still lands on Psalm 6 here — Laravel 11 app).

Two symptoms on v4.12.3 + Psalm 6.16.1:
1. `init` still writes `runTaintAnalysis="true"`, so a plain `vendor/bin/psalm` runs taint-only and reports **no type issues** (same as described above).
2. Separately, `init` now emits issue handlers that Psalm 6.16.1's config schema does not define — `MissingAbstractPureAnnotation`, `MissingPureAnnotation`, `MissingInterfaceImmutableAnnotation` — so the generated `psalm.xml` **fails to parse** out of the box: `Element '{...}MissingAbstractPureAnnotation': This element is not expected.` These look like Psalm 7 issue types; the v4 `init` template appears to target a newer Psalm than the `^6.16.1` it installs.

**Comment by alies-dev** [comment]
Correction to my previous comment — I had the version/Psalm mapping wrong. **4.x targets Psalm 7, 3.x targets Psalm 6** (as intended).

- The **`runTaintAnalysis="true"` footgun is the real, current issue**: `init` on the 3.x line (Psalm 6) writes it, so a plain `vendor/bin/psalm` runs taint-only and silently skips type analysis. Reproduced on the latest 3.x, **v3.14.2 + vimeo/psalm 6.16.1**.
- The `MissingAbstractPureAnnotation` / `MissingPureAnnotation` / `MissingInterfaceImmutableAnnotation` parse failure I mentioned is **NOT** a general bug — please disregard it. It only occurs with **v4.12.3**, which appears **mis-tagged**: it pins `vimeo/psalm ^6.16.1` (Psalm 6) while every other 4.x release requires `^7.0.0-beta19` (Psalm 7). So v4.12.3 ships 4.x code (which emits Psalm-7 issue handlers) against a Psalm-6 schema that rejects them. On a `prefer-stable` project, no-version resolution lands on this v4.12.3 anomaly. Might be worth yanking v4.12.3 from Packagist.

**Comment by alies-dev** [comment]
fixed by https://github.com/psalm/psalm-plugin-laravel/pull/1144
